test-basic/endpoint-caller.js

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2021 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var marklogic = require('../');
+var fs = require('fs');
+var testconfig = require('../etc/test-config.js');
+var expect = require('chai').expect;
+var dbWriter = marklogic.createDatabaseClient(testconfig.restWriterConnection);
+
+describe('Endpoint caller', function() {
+
+    it('moduleInitSjs endpoint', function(done) {
+        const serviceDeclaration = JSON.parse(fs.readFileSync('test-basic-proxy/ml-modules/generated/moduleInitSjs/service.json', {encoding: 'utf8'}));
+        const endpointDeclaration = JSON.parse(fs.readFileSync('test-basic-proxy/ml-modules/generated/moduleInitSjs/initializer.api', {encoding: 'utf8'}));
+        endpointDeclaration.endpoint = serviceDeclaration.endpointDirectory+'initializer'+'.sjs';
```

Comment:
A potential small refinement would be to use endpointDeclaration.functionName instead of the literal quoted string when setting the endpointDeclaration.endpoint property here and in the other tests.

That way, it's just a little easier to understand how the pieces fit together and a little easier to maintain.